src/components/search/data/searchLoader.test.jsx

```diff
@@ -124,4 +143,74 @@ describe('searchLoader', () => {
       }),
     );
   });
+
+  it('Redirect learners whose enterprise has enabled one academy.', async () => {
+    extractEnterpriseCustomer.mockResolvedValue(enterpriseCustomerFactory({ enable_one_academy: true }));
+    const academiesQuery = queryAcademiesList(mockEnterpriseCustomer.uuid);
+
+    when(mockQueryClient.ensureQueryData).calledWith(
+      expect.objectContaining({
+        queryKey: academiesQuery.queryKey,
+      }),
+    ).mockResolvedValue(mockAcademies);
+    ensureAuthenticatedUser.mockResolvedValue({ userId: 3, username: 'test-user' });
+
+    const mockParams = { enterpriseSlug: mockEnterpriseCustomer.slug };
+    const searchLoader = ({ request }) => makeSearchLoader(mockQueryClient)({ params: mockParams, request });
```

Comment:
Similar to other test cases, `makeSearchLoader(mockQueryClient)` can be passed directly to the `loader` prop. If you remove the mocking above for `react-router-dom`, React Router will pass its own `params` and `request` object without needing to manually construct/pass it yourself.